core/src/main/java/com/nukesz/game/screen/GameScreen.java

```diff
@@ -249,11 +257,31 @@ private void queryInput() {
         boolean rPressed = Gdx.input.isKeyPressed(Input.Keys.RIGHT) || Gdx.input.isKeyPressed(Input.Keys.D);
         boolean uPressed = Gdx.input.isKeyPressed(Input.Keys.UP) || Gdx.input.isKeyPressed(Input.Keys.W);
         boolean dPressed = Gdx.input.isKeyPressed(Input.Keys.DOWN) || Gdx.input.isKeyPressed(Input.Keys.S);
-
+        if (isSideControllerNeeded) {
+            if(Gdx.input.isTouched()) {
+                Vector3 touchPos = new Vector3();
+                touchPos.set(Gdx.input.getX(), Gdx.input.getY(), 0);
+                camera.unproject(touchPos);
+                if (touchPos.x<32&&touchPos.x>0){
+                    lPressed=true;
+                }
+                if (touchPos.x<WORLD_WIDTH&&touchPos.x>WORLD_WIDTH-32){
+                    rPressed=true;
+                }
+                if (touchPos.y<32&&touchPos.y>0){
+                    dPressed=true;
+                }
+                if (touchPos.y<WORLD_HEIGHT&&touchPos.y>WORLD_HEIGHT-32){
+                    uPressed=true;
+                }
+            }
+        }
```

Comment:
This part about touch (and mouse clicks maybe) probably should be in a different (helper) class. Like an InputHandler.